mongowner/mongowner-macros/src/lib.rs

```diff
@@ -37,28 +38,40 @@ pub fn derive_schema(input: TokenStream) -> TokenStream {
     let curr_struct_type = syn::Ident::new(&curr_struct, proc_macro2::Span::call_site());
 
     let fields = extract_fields_from_schema(input);
-    if let Some(_fields) = fields {
+    if let Some(fields) = fields {
         // TODO: handle the case where there is NO owned_by annotation, i.e. data subject
         // curent approach of just unwrapping causes a panic since we might be unwrapping a None
         // object when there is no owned_by annotation
-        //
-        // let field = find_fk_field(&fields).unwrap();
-        // let owner_type_ident = {
-        //     let owner_type = find_owner(&field).unwrap();
-        //     syn::Ident::new(&owner_type, proc_macro2::Span::call_site())
-        // };
-        // let fk_field_name = {
-        //     match &field.ident {
-        //         Some(ident) => ident.to_string(),
-        //         None => "".to_string()
-        //     }
-        // };
+        println!("curr_struct: {}", curr_struct);
+        let mut owner_type_ident: Option<Ident> = None;;
+        for field in fields.named.iter() {
+            let fk_field_name = {
+                match &field.ident {
+                    Some(ident) => ident.to_string(),
+                    None => "".to_string()
+                }
+            };
+            println!("fk_field_name: {}", fk_field_name);
+            let owner_type_str = find_owner(&field).unwrap_or_else(|| {
+                "".to_string()
+            });
+            if(owner_type_str != "") {
+                owner_type_ident = Some(syn::Ident::new(&owner_type_str, proc_macro2::Span::call_site()));
+            }  
+        }
+        let owner_type = owner_type_ident.unwrap_or_else(
+            || panic!("No owner type found for {}", curr_struct)
```

Comment:
We should probably do something here where we check if it's a data subject/"root owner" or something rather than just panicking since the User will be a #Schema but won't necessarily have a owner. Do you want me to handle it? I was going to work on making the parser more reusable and generic

Reply:
Yeah, that works! Another idea: what if we just made the `User` have an owner? See https://github.com/naafiyan/mongodog/pull/7/files#diff-aa92b17a5bc36a7f400de988e97118b0141b4ce85badfde04a8fdefe8b49a660

Reply:
ie User is owned by itself

Reply:
cc @swetabhch